org.scala-refactoring.library/src/main/scala/scala/tools/refactoring/common/Selections.scala

```diff
@@ -85,12 +85,27 @@ trait Selections extends TreeTraverser with common.PimpedTrees {
      * fully contains a SymTree, if true, the first selected is returned. Otherwise
      * the result of findSelectedOfType[SymTree] is returned.
      */
-    lazy val selectedSymbolTree = (root filter (cond(_) {
+    lazy val selectedSymbolTree = eventuallyFixModifierPositionsForLazyVals((root filter (cond(_) {
       case t: SymTree => contains(t)
     }) filter (t => t.pos.start < t.pos.end) match {
       case (x: SymTree) :: _ => Some(x)
       case _ => None
-    }) orElse findSelectedOfType[SymTree]
+    }) orElse findSelectedOfType[SymTree])
+
+    /*
+     * See #1002392 if you wonder why we need this
+     */
+    private def eventuallyFixModifierPositionsForLazyVals(t: Option[SymTree]): Option[SymTree] = t.map {
+      case lDef: DefDef if lDef.mods.isLazy && lDef.mods.positions.isEmpty =>
+        root.foreach {
+          case lVal: ValDef if lDef.mods.isLazy && !lVal.mods.positions.isEmpty && lVal.pos.point == lDef.pos.point =>
+            lDef.mods.setPositions(lVal.asInstanceOf[ValDef].mods.positions)
```

Comment:
I too think that it would be safer to copy the DefDef (it has a copy method, and you can use setSymbol and setType to assign these) instead of modifying it because we don't create it ourselves but get it from the compiler.